sparse/core.py

```diff
@@ -245,26 +245,43 @@ def __sizeof__(self):
 
     def __getitem__(self, index):
         if not isinstance(index, tuple):
-            index = (index,)
+            if isinstance(index, str):
+                data = self.data[index]
+                idx = np.where(data)
+                coords = []
+
+                for i in range(self.ndim):
+                    coords.append(self.coords[i, idx[0]])
+
+                for i in range(1, np.ndim(data)):
+                    coords.append(idx[i])
```

Comment:
I don't understand the purpose of this.  Can you help explain?

Reply:
Custom `dtype`s can have multiple dimensions inside a single scalar 'field'. For example, `np.dtype([('grades', np.float64, (2, 2))])` will have a `(2, 2)` element "inside" each "scalar" element. `self.data['grades']` gets a `(nnz, 2, 2)` array in this case, but the resulting array has to be `self.shape + (2, 2)`. We first use `np.where` to get the nonzero indices. Then we append the matching parts of the original coordinates above, we get those from `idx[0]`. After this, we get the parts that come from the dimensions of the field itself, and append those to `coords`.

Finally, we flatten the data since we're done calculating all the `coords` of the resulting array.

Reply:
Right, I see.  I had expected the dtype to continue being of shape `(2, 2)` but I see that NumPy rolls this into the normal array's shape

```python

In [1]: import numpy as np

In [2]: np.ones((5,), dtype=np.dtype([('grades', np.float64, (2, 2))]))
Out[2]: 
array([([[ 1.,  1.], [ 1.,  1.]],), ([[ 1.,  1.], [ 1.,  1.]],),
       ([[ 1.,  1.], [ 1.,  1.]],), ([[ 1.,  1.], [ 1.,  1.]],),
       ([[ 1.,  1.], [ 1.,  1.]],)],
      dtype=[('grades', '<f8', (2, 2))])

In [3]: x = np.ones((5,), dtype=np.dtype([('grades', np.float64, (2, 2))]))

In [4]: x.shape
Out[4]: (5,)

In [5]: x['grades'].shape
Out[5]: (5, 2, 2)

In [6]: x['grades'].dtype
Out[6]: dtype('float64')

```